drivers/w1/w1_max32.c

```diff
@@ -0,0 +1,207 @@
+/*
+ * Copyright (c) 2023-2024 Analog Devices, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/drivers/w1.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/drivers/pinctrl.h>
+#include <zephyr/drivers/clock_control/adi_max32_clock_control.h>
+
+#include <wrap_max32_owm.h>
+
+#define DT_DRV_COMPAT adi_max32_w1
+
+LOG_MODULE_REGISTER(w1_max32, CONFIG_W1_LOG_LEVEL);
+
+struct max32_w1_config {
+	struct w1_master_config w1_config;
+	mxc_owm_regs_t *regs;
+	const struct pinctrl_dev_config *pctrl;
+	const struct device *clock;
+	struct max32_perclk perclk;
+	int internal_pullup;
+	int external_pullup;
+	int long_line_mode;
+};
+
+struct max32_w1_data {
+	struct w1_master_data w1_data;
+	uint8_t reg_device_config;
+};
+
+static int api_reset_bus(const struct device *dev)
+{
+	int ret;
+	const struct max32_w1_config *cfg = dev->config;
+	mxc_owm_regs_t *regs = cfg->regs;
+
+	/* 0 if no 1-wire devices reponded during the presence pulse, 1 otherwise */
+	ret = MXC_OWM_Reset();
+	if (ret > 0) {
+		/* Check OW pin input state due to presence detect pin seems not work well */
+		if (regs->ctrl_stat & MXC_F_OWM_CTRL_STAT_OW_INPUT) {
+			ret = 1; /* At least 1 device on the line */
+		} else {
+			ret = 0; /* no device on the line */
+		}
+	}
+
+	return ret;
+}
+
+static int api_read_bit(const struct device *dev)
+{
+	int ret;
+
+	ret = MXC_OWM_ReadBit();
+	if (ret < 0) {
+		if (MXC_OWM_GetPresenceDetect() == 0) {
+			/* if no slave connected to the bus, read bits shall be logical ones */
+			ret = 1;
+		} else {
+			return -EIO;
+		}
+	}
+
+	return ret;
+}
+
+static int api_write_bit(const struct device *dev, bool bit)
+{
+	int ret;
+
+	ret = MXC_OWM_WriteBit(bit);
+	if (ret < 0) {
+		if (MXC_OWM_GetPresenceDetect() == 0) {
+			/* if no slave connected to the bus, write shall success */
+			ret = 0;
+		} else {
+			return -EIO;
+		}
+	}
+
+	return ret;
+}
+
+static int api_read_byte(const struct device *dev)
+{
+	int ret;
+
+	ret = MXC_OWM_ReadByte();
+	if (ret < 0) {
+		if (MXC_OWM_GetPresenceDetect() == 0) {
+			/* if no slave connected to the bus, read bits shall be logical ones */
+			ret = 0xff;
+		} else {
+			return -EIO;
+		}
+	}
+
+	return ret;
+}
+
+static int api_write_byte(const struct device *dev, uint8_t byte)
```

Comment:
You could change the prefix of `api_` functions to `w1_max32_`. I had a similar feedback in the past.

https://github.com/zephyrproject-rtos/zephyr/pull/76627#discussion_r1702223265

Reply:
Thanks but, there are too many w1 and max32 inside the code, so we decided to add api_ prefix for all drivers to get more readable driver code file.
Existing uart/ gpio... has same format, I prefer to keep it as it is.